packages/volto/src/helpers/Url/Url.js

```diff
@@ -94,14 +100,30 @@ export function getView(url) {
  * @returns {string} Flattened URL to the app server
  */
 export function flattenToAppURL(url) {
+  let adjustedUrl = url;
   const { settings } = config;
-  return (
-    url &&
-    url
+  const prefix = settings.prefixPath;
+
+  adjustedUrl =
+    adjustedUrl &&
+    adjustedUrl
       .replace(settings.internalApiPath, '')
       .replace(settings.apiPath, '')
-      .replace(settings.publicURL, '')
-  );
+      .replace(settings.publicURL, '');
+  const internalURL =
+    adjustedUrl &&
+    (adjustedUrl.indexOf(settings.publicURL) !== -1 ||
+      (settings.internalApiPath &&
+        adjustedUrl.indexOf(settings.internalApiPath) !== -1) ||
+      adjustedUrl.indexOf(settings.apiPath) !== -1 ||
+      adjustedUrl.charAt(0) === '/' ||
+      adjustedUrl.charAt(0) === '.');
+  //using isInternalUrl method causes infinite loop with special externalRoutes defined in wise-marine #264955
+  if (internalURL && !adjustedUrl.startsWith('#')) {
```

Comment:
@nileshgulia1  @sneridagh Can `flattenToAppURL` receive external URLs or full URLs from the portal itself?

`internalURL` will never start with `#`.